Sources/TuistAutomation/ProjectMappers/SourceRootPathProjectMapper.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistGraph
+
+/// Automation commands create their own project in temporary directory
+/// This means `SRCROOT` has a different path from the directory where `.xcodeproj` resides
+/// `SRCROOT` should point to the directory of the target sources
+/// To ensure projects still build, we need to overwrite `SRCROOT` variable ourselves
+/// This should be used only for automation projects as they are not expected to have their location changed after generation
+public final class SourceRootPathProjectMapper: ProjectMapping {
+    public init() {}
+
+    public func map(project: Project) throws -> (Project, [SideEffectDescriptor]) {
+        var project = project
+        var base = project.settings.base
+        // Keep the value if defined by user
+        if base["SRCROOT"] == nil {
```

Comment:
This doesn't prevent developers from overriding the value at the target level. In that case, the selective testing wouldn't work. I wonder if we should set that value at the target level instead and ignore whatever value is set. Since this mapper is only used for selective testing, I think it's fine. We can print a log telling developers that we are doing that.

Reply:
> This doesn't prevent developers from overriding the value at the target level.

I believe it's fine if users override this value. The original issue is that `SRCROOT` was equal to `PROJECT_DIR` but in this case it would stay unchanged. So if your "normal" project builds, so should the one for selective testing as the `SRCROOT` will be the same for both

Reply:
@pepibumur does this make sense to you or are you still of the opinion that we should change this?

Reply:
It does! Thanks for clarifying it.